src/Lib/Canvas.vala

```diff
@@ -140,6 +140,8 @@ public class Akira.Lib.Canvas : Goo.Canvas {
     }
 
     public override bool button_press_event (Gdk.EventButton event) {
+        focus_canvas ();
+
```

Comment:
Do we need to set the focus on the canvas if a button is pressed?
Isn't the very own action of clicking on the canvas moving the focus there?

Reply:
Yes, clicking on the canvas moves the focus on the canvas, but not the "keyboard" focus, so we need to restore this focus on the canvas in order to listen again for the keyboard related actions (delete, insert, etc). This is especially true when use put text into the enties. After your insertion the Canvas "naturally" doesn't receive back the keyboard focus, so we need to force the focus update.

Reply:
Sounds good, thanks for the explanation.